eZ/Bundle/EzPublishCoreBundle/Resources/config/image.yml

```diff
@@ -83,6 +84,13 @@ services:
         tags:
             - { name: liip_imagine.cache.resolver, resolver: ezpublish }
 
+    ezpublish.image_alias.imagine.cache_resolver_decorator:
+        class: '%liip_imagine.cache_resolver.proxy.class%'
+        decorates: ezpublish.image_alias.imagine.cache_resolver
+        arguments:
+            - '@ezpublish.image_alias.imagine.cache_resolver_decorator.inner'
+            - '$image_hosts$'
```

Comment:
I had a configuration like this:
```
siteaccess:
        list:
          - site
          - eng
        groups:
            site_group:
                - site
                - eng
        default_siteaccess: site
        match:
            URIElement: 1
```
and
```
system:
        default:
            image_hosts: ['http://default.com']
        site_group:
            image_hosts: ['http://sitegroup.com']
        site:
            image_hosts: ['http://site.com']
        eng:
            image_hosts: ['http://eng.com']
```

I have to admit that you are right. Apparently, something is wrong.

URL | SiteAccess | Image URL
---- | ----------- | -----------
test.ez | site | sitegroup.com 
test.ez/eng | eng | sitegroup.com
test.ez/site | site | sitegroup.com

When I remove:
```
        site_group:
            image_hosts: ['http://sitegroup.com']
```
from the configuration, then result is:

URL | SiteAccess | Image URL
---- | ----------- | -----------
test.ez | site | default.com 
test.ez/eng | eng | eng.com
test.ez/site | site | default.com

When I remove:
```
        default:
            image_hosts: ['http://default.com']
```
from the configuration, then result is:

URL | SiteAccess | Image URL
---- | ----------- | -----------
test.ez | site | site.com 
test.ez/eng | eng | eng.com
test.ez/site | site | site.com

@alongosz any insight how to fix that? It is related to the configuration merging issue, isn't it?

Reply:
@kmadejski there are two separate issues here.
1. If default settings exist, when container is compiled, these settings are injected by default, because at that time there is no SA context yet. Workarounds:
    - add `lazy: true`
    - don't inject it directly, inject `@ezpublish.config.resolver` instead and get parameter when needed (then context should be already set).
2. In case of arrays and multiple scope, the merging is done incorrectly. This is more general issue (e.g. happens for `languages` as well). We've discussed it several times and we're worried that changing this behavior now would be a BC break. The issue exists in `eZ\Bundle\EzPublishCoreBundle\DependencyInjection\Configuration\SiteAccessAware\Contextualizer::mapConfigArray` in:
```php
$mergedSettings = array_merge(
                    $defaultSettings,
                    $groupsSettings,
                    $scopeSettings,
                    $globalSettings
                );
```
The problem is that for scalars this works as expected - the last one is the one set, however in case of arrays merging merges the first one on top, which is rather not desired.

So for 1. you could either try one of the workarounds (though lazy doesn't work for me in your particular case) or you could investigate how to improve dynamic settings injection so the issue disappears. We could really benefit from this.

For 2. I think this is out of scope of the PR, however suggestion on how to solve it is welcomed.

For 1. & 2. you could play with normalization of semantic config for `arrayNode('image_hosts')`, but this is just an educated guess - you'd need to investigate it :)